### ISSUE TITLE
"cdk diff" should take assets into account

### ISSUE THREAD

**Issue by eladb** [aws/aws-cdk#395]
When running `cdk diff`, the toolkit should take assets into account. At the moment, if only assets change, "cdk diff" will show no differences, but "cdk deploy" will update the stack since the asset's md5 changes.

**Comment by brad-jones** [comment]
For a quick and dirty hack I just inject a hash of the asset into the stack somewhere, in the case of a lambda just set an environment variable. Then when running `cdk diff` it's obvious what will update.

```typescript
new lambda.Function(this, 'SomeLambda', {
	code: lambda.Code.asset('/path/to/zip'),
	environment: {
		HASH: HashFuncOfYourChoice('/path/to/zip'),
	},
	etc...
});
```

**Comment by eladb** [comment]
Copy @sam-goodwin

**Comment by barroij** [comment]
Having the same issue, but as I'm not referencing a zip file but a directory, I cannot use @brad-jones workaround.

The zip md5 could be made available so that I can put its value an an output value (supposing that different output values buiolt with `new Cdk.CfnOutput` sould cause the stack to be deployed)

**Comment by jeffsee55** [comment]
Wonder if this could be achieved by using a checksum? That would have the benefit of https://github.com/aws/aws-cdk/issues/395#issuecomment-474724435 's comment without having to zip beforehand. 

Since the `lambda.Code.fromAsset(...`, would know the directory it may even be possible to perform automatically and store in metadata?

**Comment by eladb** [comment]
will be fixed by #2821
